RecoLuminosity/LumiProducer/test/analysis/plugins/PCCNTupler.cc

```diff
@@ -0,0 +1,287 @@
+                pileup->Fill(PVI->getPU_NumInteractions());
+            }
+        }
+    }
+
+    // Get the Run, Lumi Section, and Event numbers, etc.
+    runNo = iEvent.id().run();
+    LSNo = iEvent.getLuminosityBlock().luminosityBlock();
+    LNNo = -99; // FIXME need the luminibble
+    eventNo = iEvent.id().event();
+    timeStamp = iEvent.time().unixTime();
+   
+    if((BXNo.count(iEvent.bunchCrossing())==0||nGoodVtx.count(iEvent.bunchCrossing())==0) && !(BXNo.count(iEvent.bunchCrossing())==0&&nGoodVtx.count(iEvent.bunchCrossing())==0)){
+        std::cout<<"BXNo and nGoodVtx should have the same keys but DO NOT!!!"<<std::endl;
+    }
+    
+    if(BXNo.count(iEvent.bunchCrossing())==0){
+        BXNo[iEvent.bunchCrossing()]=0;
+    }
+
+    if(nGoodVtx.count(iEvent.bunchCrossing())==0){
+        nGoodVtx[iEvent.bunchCrossing()]=0;
+    }
+
+    BXNo[iEvent.bunchCrossing()]=BXNo[iEvent.bunchCrossing()]+1;
+    // add the vertex information
+
+    xV = -9999.; yV = -9999.; zV = -9999.;  chi2 = -9999.;  ndof = -9999;
+   
+    if(includeVertexInformation){
+        edm::Handle<reco::VertexCollection> recVtxs;
+        iEvent.getByLabel("offlinePrimaryVertices",recVtxs);
+   
+        if(recVtxs.isValid()){
+            for(reco::VertexCollection::const_iterator v=recVtxs->begin(); v!=recVtxs->end(); ++v){
+                int nnTrk = v->tracksSize();
+                int nd = (int)v->ndof();
+                if(nd > 4 && v->isValid() && (v->isFake() == 0)){
+                    nVtx++;
+                    if(nnTrk > 0){
+                        nGoodVtx[iEvent.bunchCrossing()]=nGoodVtx[iEvent.bunchCrossing()]+1;
+                        if(nnTrk > nTrk){
+                            nTrk = nnTrk;                   
+                            //FIXME why are we multiplying by 10000 or 10?
+                            xV = 10000.*v->x(); 
+                            yV = 10000.*v->y(); 
+                            zV = 10.*v->z();
+                            chi2 = v->chi2();
+                            ndof = (int)v->ndof();
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+    // -- Does this belong into beginJob()?
+    ESHandle<TrackerGeometry> TG;
+    iSetup.get<TrackerDigiGeometryRecord>().get(TG);
+
+    // -- FED
+    edm::eventsetup::EventSetupRecordKey recordKey(edm::eventsetup::EventSetupRecordKey::TypeTag::findType("RunInfoRcd"));
+    if (recordKey.type() == edm::eventsetup::EventSetupRecordKey::TypeTag()) {
+        //record not found
+        cout << "Record \"RunInfoRcd" << "\" does not exist " << endl;
+        fFED1 = 0; 
+        fFED2 = (0x1 << 12) ; 
+    } else {
+        edm::ESHandle<RunInfo> runInfoHandle;
+        iSetup.get<RunInfoRcd>().get(runInfoHandle);
+    }
+    // -- Pixel cluster
+    edm::Handle< edmNew::DetSetVector<SiPixelCluster> > hClusterColl;
+    iEvent.getByLabel(fPixelClusterLabel, hClusterColl);
```

Comment:
Do you mean just lines 236 or anytime I'm potentially accessing a collection from data?  If just for data then the pileup info is ok as it is, right?